### PR TITLE
Making test run

### PR DESCRIPTION
**Pull request by jamestwebber**
I added a new script `make_test_cbcls.py` which can produce a subset of a NovaSeq run with a much smaller number of bases, tiles, and reads.

Appears to work correctly on the `190414_A00111_0296_AHJCWWDSXX` dataset from before–I made a 24-base run (including indexes) from the first 100 reads of the first 3 tiles, and when I extracted it via Python it produced the correct output. The total data size is less than a megabyte, so hopefully this should allow some rapid testing.

I also moved the `test_data` folder to the top level instead of having it in `src` because I think that's cleaner. I tried to modify paths where I could find them but might have missed a few.

### CODE REVIEW

**Comment by jamestwebber** [comment]
Oh, and maybe we can get rid of some of the other stuff in `test_data`, but I'll leave that up to you because you were using it before.

**Comment by jamestwebber** [comment]
Also: the script I wrote requires the [`seqbot`](https://www.github.com/czbiohub/seqbot) library because it seemed silly to repeat all of that code. That pulls in a few other requirements (like `click` and `numpy`).

At this point it wouldn't be crazy to move all of the python `bcl2fastr` code into this repo, but I'm not sure how to structure a combination Rust/Python package.

**Comment by phoenixAja** [comment]
@jamestwebber yeah i agree with you that we should move over the python `bcl2fastr` stuff over.. let me look into a good way to do so that's clean. I think that can be a seperate PR/issue though so we can merge this in soon.

File: src/filter_decoder.rs [hunk 1]
```diff
@@ -1,59 +1,59 @@
...
-                    false,
-                    false,
-                    true,
-                    true,
-                ],
-            };
-        assert_eq!(actual_filter, expected_filter);
-    }
-}
+use byteorder::{LittleEndian, ReadBytesExt};
```
Comment (phoenixAja) [inline src/filter_decoder.rs]: was this file copy and pasted? it's hard to see where you made changes.

File: src/locs_decoder.rs [hunk 2]
```diff
@@ -1,76 +1,76 @@
...
-                    vec![
-                        7.232394,
-                        0.0,
-                    ],
-                ],
-            };
-        assert_eq!(actual_locs, expected_locs)
-    }
-}
+use byteorder::{LittleEndian, ReadBytesExt};
```
Comment (phoenixAja) [inline src/locs_decoder.rs]: same here about being copied, i might be missing something.

File: src/main.rs [hunk 3]
```diff
@@ -13,22 +13,22 @@ mod cbcl_header_decoder;
...
     // filter_decoder::filter_decoder(filter_file);
 
-    let locs_path = Path::new("src/test_data/test_locs.locs");
-    let run_info_path = Path::new("src/test_data/RunInfo.xml");
-    let run_params_path = Path::new("src/test_data/test_runparams.xml");
-    let lane_path = Path::new("src/test_data/test_lane/");
+    let locs_path = Path::new("test_data/test_locs.locs");
+    let run_info_path = Path::new("test_data/RunInfo.xml");
+    let run_params_path = Path::new("test_data/test_runparams.xml");
+    let lane_path = Path::new("test_data/test_lane/");
```
Comment (phoenixAja) [inline src/main.rs]: thanks for changing these!

File: src/filter_decoder.rs (hunk 1 above)
Comment (jamestwebber) [reply src/filter_decoder.rs]: Not sure what happened to the diff, it looks fine for my locally. The only change was line 45, removed `src` from the test data path

File: src/locs_decoder.rs (hunk 2 above)
Comment (jamestwebber) [reply src/locs_decoder.rs]: Oh I wonder if the file encoding changed or something like that?

**Comment by phoenixAja** [comment]
give me some time to run this through locally and then I'll merge it in 👍

File: src/locs_decoder.rs (hunk 2 above)
Comment (jamestwebber) [reply src/locs_decoder.rs]: again, just removed `src` from the test data path

File: src/filter_decoder.rs (hunk 1 above)
Comment (phoenixAja) [reply src/filter_decoder.rs]: weird, i figured something strange was going on.. I think it's fine for now

Comment (phoenixAja) [reply src/filter_decoder.rs]: you might be accidentally blamed for things in the future though 😄

**Comment by jamestwebber** [comment]
When merged this closes #4 and #35

**Pull request merged**